packages/nx-plugin-unused-deps/src/generators/check/generator.ts

```diff
@@ -25,12 +24,12 @@ export default async function (tree: Tree, options: CheckGeneratorSchema) {
   const packageJson = readJson<PackageJson>(tree, 'package.json');
   const jsonFile = `.nx-plugin-unused-deps.json`;
 
-  const filterPackageRecord = (depName) => !packageRecord[`npm:${depName}`];
+  const packageIsNotDependedOn = (depName: string) =>
+    !packageRecord[`npm:${depName}`] && !packageRecord[`npm:${depName.replace('@types/', '')}`];
```

Comment:
`depName` is the name of a package listed in package.json's dependencies or devDependencies, `packageRecord` is the list of dependencies as generated by the nx graph. the nx graph will never have `npm:@types/X` listed as a dependency, it will have `npm:X` as a dependency, but we want to match `depName` of `X` and `@types/X` to say that both are depended on if `npm:X` is a dependency.

Reply:
that said, the logic might be more obvious if there was a `startsWith('@types/')` ternary rather than just an `&&`